oauth-common/src/main/java/io/strimzi/kafka/oauth/common/BearerTokenWithPayload.java

```diff
@@ -25,6 +27,10 @@
 
     void setPayload(Object payload);
 
+    List<String> getGroups();
```

Comment:
Isn't it logically a `Set`?

Reply:
I am unsure about this one as well.

That depends on the semantics. Mostly people will want it to have a Set semantics, but the authorization server may maintain predictable order, and in principle it may allow configuring the order (I don't know how common that is). In such a case, since JSON array or a string is always used to communicate the info, the ordering is maintained all the way to the BearerTokenWithPayload object, and could possibly be used when making authorization decision in a custom authorizer. While I have no useful use-case in mind, maybe someone else has - at their peril. Thus, I see this as an additional degree of freedom, that shouldn't really hurt anyone.

Reply:
Since most people I talked to expect a Set here I'm changing to Set.